libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/NavigationInstructionPlayer.java

```diff
@@ -35,44 +40,41 @@ private void initAudioFocusRequest() {
     }
   }
 
-  private void initInstructionPlayer(Context context, @Nullable String awsPoolId,
-                                     Locale locale) {
-    if (!TextUtils.isEmpty(awsPoolId)) {
-      instructionPlayer = new PollyPlayer(context, awsPoolId, locale);
-      isPollyPlayer = true;
-    } else {
-      instructionPlayer = new DefaultPlayer(context, locale);
-    }
-    instructionPlayer.addInstructionListener(this);
+  private void initInstructionPlayers(Context context, Locale locale) {
+    mapboxSpeechPlayer = new MapboxSpeechPlayer(context, locale, this);
+    androidSpeechPlayer = new AndroidSpeechPlayer(context, locale, this);
   }
 
-  @Override
-  public void play(String instruction) {
-    instructionPlayer.play(instruction);
+  public void play(@NonNull String instruction, @Nullable VoiceInstructionMilestone voiceInstructionMilestone) {
+    instructionQueue.add(Pair.create(instruction, voiceInstructionMilestone));
```

Comment:
Love the use of `Queue` here - `VoiceInstructionMilestone` has both the plain text and SSML versions of he instruction.  Would using `VoiceInstructionMilestone#getAnnouncement()` here remove the need to create these pairs?

Reply:
@danesfeder the weird case where we need the String is this https://github.com/mapbox/mapbox-navigation-android/pull/751/files/f0641f159265aad6013ddf7adcef18743622ead0#diff-72d1cc8dfc72651371814beec178aecbR70